### PR TITLE
Open IoT SDK port

### PR DESCRIPTION
**Pull request by ATmobica**
Open IoT SDK IP review

### CODE REVIEW

File: src/system/SystemMutex.h [hunk 1]
```diff
@@ -155,5 +160,14 @@ inline void Mutex::Unlock(void)
 }
 #endif // CHIP_SYSTEM_CONFIG_FREERTOS_LOCKING
 
+#if CHIP_SYSTEM_CONFIG_CMSIS_RTOS_LOCKING
+inline CHIP_ERROR Mutex::Init(Mutex & aMutex)
+{
+    // The mutex is initialized when constructed and generates
```
Comment (pan-) [inline src/system/SystemMutex.h:166]: Why not delay the initialisation to this point ? It is easier to catch errors.

File: src/controller/tests/TestEventChunking.cpp [hunk 2]
```diff
@@ -542,10 +542,10 @@ nlTestSuite sSuite =
 
 } // namespace
 
-int TestReadChunkingTests()
+int TestEventChunkingTests()
```
Comment (pan-) [inline src/controller/tests/TestEventChunking.cpp:545]: What was wrong with this test ? duplicate names ?

File: src/transport/raw/tests/NetworkTestHelpers.cpp [hunk 3]
```diff
@@ -29,6 +29,10 @@ namespace Test {
 
 CHIP_ERROR IOContext::Init()
 {
+#if defined(CHIP_DEVICE_LAYER_TARGET_OPEN_IOT_SDK)
```
Comment (pan-) [inline src/transport/raw/tests/NetworkTestHelpers.cpp:32]: Was this error due to the chip stack being already initialized ?

File: .github/workflows/examples-openiotsdk.yaml [hunk 4]
```diff
@@ -35,7 +35,10 @@ jobs:
         if: github.actor != 'restyled-io[bot]'
 
         container:
-            image: connectedhomeip/chip-build-openiotsdk:0.5.101
+            image: atmobica/chip-build-openiotsdk:latest
```
Comment (pan-) [inline .github/workflows/examples-openiotsdk.yaml:38]: We will have to remove this before upstream.

File: .devcontainer/devcontainer.json [hunk 5]
```diff
@@ -17,7 +17,7 @@
     "build": {
         "dockerfile": "Dockerfile",
         "args": {
-            "BUILD_VERSION": "0.5.99"
+            "BUILD_VERSION": "0.5.101"
```
[the remark this answers is not in the document]
Comment (ATmobica) [reply .devcontainer/devcontainer.json:20]: Oh, my oversight. Anyway, during rebase I'll need to fit it to right value, now  it 0.6.01.

File: src/controller/tests/TestEventChunking.cpp (hunk 2 above)
Comment (ATmobica) [reply src/controller/tests/TestEventChunking.cpp:545]: Yes, a mismatch between the file name and test and TestEventChunkingTests already exists.

File: .github/workflows/examples-openiotsdk.yaml (hunk 4 above)
Comment (ATmobica) [reply .github/workflows/examples-openiotsdk.yaml:38]: Sure

File: src/transport/raw/tests/NetworkTestHelpers.cpp (hunk 3 above)
Comment (ATmobica) [reply src/transport/raw/tests/NetworkTestHelpers.cpp:32]: The issue was that we needed to initialize the stack and SDK timer for the correct operation of the tests.

**Review by paul-szczepanek-arm — APPROVED** [review]

**Review by pan- — APPROVED** [review]

**Comment by paul-szczepanek-arm** [comment]
is it OK to commit symlinks to the repo? does chip have symlinks anywhere already?

**Comment by ATmobica** [comment]
> is it OK to commit symlinks to the repo? does chip have symlinks anywhere already?
Yes, they love symbolic links :)

**Pull request merged**